qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -502,6 +502,13 @@ class WebEngineHistory(browsertab.AbstractHistory):
     """QtWebEngine implementations related to page history."""
 
     def current_idx(self):
+        if self.to_load:
+            idx = next(
```

Comment:
do you mean doing something like?:

    try:
        idx = [i for i, item in enumerate(self.to_load) if item.active][0]
    except IndexError:
        idx = len(self.to_load) - 1

I used next() to make it stop iterating as soon as it finds the active item, but that might be a premature optimisation. And it would probably be more effecient to iterate in reversed order since the active item is more likely to be near the end of the list.

Reply:
Oh okay, I hadn't seen that pattern before.